h2o-core/src/main/java/water/api/SchemaMetadata.java

```diff
@@ -203,8 +204,9 @@ public static FieldMetadata createIfApiAnnotation(Schema schema, Field f, List<F
 
       if (null != f.getAnnotation(API.class))
         return new FieldMetadata(schema, f, superclassFields);
-
-      Log.warn("Skipping field that lacks an annotation: " + schema.toString() + "." + f);
+      if (!(Modifier.isPrivate(f.getModifiers()))) { // don't warn in case of private field without annotation
```

Comment:
why is this necessary?

Reply:
it is because of ModelSchemaV3.java#input_parameters field. It is a private field and it doesn't have to be exposed in the API. It is used only in `ModelParametersSchemaV3.writeParametersJSON(ab, parameters, input_parameters, defaults ,"parameters");` But when I just remove @API notation, INFO check starts failing, because of the field without annotation found. So I can either expose what is not needed to be exposed or modify this check to not warn on fields without annotation when they are private. Private fields are also pretty rare in schemas, we have some only in TimelineV3. These have api annotation, so they won't be affected by this change. Do you think it would be better to expose and leave this check as it was?

Reply:
Those private fields in TimelineV3 would be affected, because if their annotation was accidentally removed no warning message would be logged. 
I am having hard time coming up with a good solution, only thing that comes to mind is to have some @Ignore annotation, but thats also not very pretty.
Another thought you say it is not exposed in the API, yet in writeParametersJSON you write out the value into the JSON so in fact it is exposed in the API and should be public, problem solved.

Reply:
sounds good public

Reply:
@honzasterba do you know why those Timelinev3 fields are even private? if they're part of the API? 
I don't think we should make things more complex only because of an old decision (private fields that should be public) that we don't understand or that doesn't  seem to make sense.
Is it not this class that should have all its API fields public?

Making `input_parameters` public (with an API annotation) is also nonsense , because it is not exposed as such, but only as the `input` field in for each param, and we don't want to expose those values twice.

Reply:
@sebhrusen  Sorry I have not noticed this is about a field on ModelSchemaV3, you are correct

AFAIK schema does not differentiate between public/private so we could fix TimelineV3 and fix this check to only apply to non-anotated public fields and also make Schema only check public fields

Reply:
Yes, for consistency, I'd also fix `TimelineV3` to use public fields. I think this is the only use case in the code base.
Did see any other in `core`, `algos` or `automl`.

Reply:
So... will leave ModelSchemaV3.java and also binding.py as they are, so that we have input_parameters only in context where we want them to be. And change private fields in TimelineV3 to public so that if their annotation was accidentally removed we will get warning. We have also some protected fields and fields without explicit modifier.. So I would not warn only on private non-anotated fields and warn on all the others non-anotated (as it is now in 02884ea) Allright?